src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Legacy/HtmlMarkupParser.cs

```diff
@@ -1085,16 +1085,50 @@ private void ParseAttribute(in SyntaxListBuilder<RazorSyntaxNode> builder)
         // http://dev.w3.org/html5/spec/tokenization.html#attribute-name-state
         // Read the 'name' (i.e. read until the '=' or whitespace/newline)
         using var nameTokens = new PooledArrayBuilder<SyntaxToken>();
-        if (!TryParseAttributeName(ref nameTokens.AsRef()))
+        if (TryParseAttributeName(out SyntaxToken? ephemeralToken, ref nameTokens.AsRef())
```

Comment:
This is a minor nit, but couldn't all of the non-success cases be handled in the switch to simplify the conditional logic?

```C#
var nameParsingResult = TryParseAttributeName(out SyntaxToken? ephemeralToken, ref nameTokens.AsRef());

// Handle non-AttributeNameParsingResult.Success cases first (each one exits).
switch (nameParsingResult)
{
    // Parse C# and return to attribute parsing afterwards.
    case AttributeNameParsingResult.CSharp:
    {
        Accept(in attributePrefixWhitespace);
        PutCurrentBack();
        using var pooledResult = Pool.Allocate<RazorSyntaxNode>();
        var dynamicAttributeValueBuilder = pooledResult.Builder;
        OtherParserBlock(dynamicAttributeValueBuilder);
        var value = SyntaxFactory.MarkupMiscAttributeContent(dynamicAttributeValueBuilder.ToList());
        builder.Add(value);
        return;
    }

     // Parse razor comment and return to attribute parsing afterwards.
    case AttributeNameParsingResult.RazorComment:
    {
        Accept(in attributePrefixWhitespace);
        PutCurrentBack();
        ParseRazorCommentWithLeadingAndTrailingWhitespace(builder);
        return;
    }

    // Unexpected character in tag, enter recovery
    case AttributeNameParsingResult.Other:
    {
        Accept(in attributePrefixWhitespace);
        ParseMiscAttribute(builder);
        return;
    }
}
```